irdb/publish.py

```diff
@@ -1,60 +1,57 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
 """Publish and upload irdb packages"""
-import sys
-import shutil
+
+import argparse
+import logging
+import getpass
+from typing import Optional
+from warnings import warn
 from pathlib import Path
-from tempfile import TemporaryDirectory
 from datetime import datetime as dt
+from zipfile import ZIP_DEFLATED, ZipFile
+
 import yaml
 import pysftp
 
+try:
+    from .publish_utils import _is_stable, get_stable, get_all_package_versions
+except ImportError:
+    from publish_utils import _is_stable, get_stable, get_all_package_versions
+
 PATH_HERE = Path(__file__).parent
 PKGS_DIR = PATH_HERE.parent
-OLD_FILES = PKGS_DIR / "_OLD_FILES"
 ZIPPED_DIR = PKGS_DIR / "_ZIPPED_PACKAGES"
 
-PATH_PACKAGES_YAML = PATH_HERE / "packages.yaml"
-
-SERVER_DIR = PATH_HERE / "InstPkgSvr"
-
-HELPSTR = r"""
-Publish stable IRDB packages
-----------------------------
-This command must be run from the IRDB root directory
+PATH_FOLDERS_YAML = PATH_HERE / "server_folders.yaml"
 
-$ python irdb/publish.py -c -u <PKG_NAME> ... <PKG_NAME_N> -l <USERNAME> -p <PASSWORD>
 
--l <USERNAME> : UniVie u:space username - e.g. u\kieranl14
--p <PASSWORD> : UniVie u:space password
--d : [update-version] : do not update the version of the package to today
--c : [compile] all files in a PKG folder to a .zip archive
--cdev : [compile-dev] like compile, but tags as development version
--u : [upload] the PKG .zip archive to the server
--h : [help] prints this statement
+class Password:
+    """Used for secure pwd promt."""
+    DEFAULT = "Prompt if not specified"
 
-Arguments without a "-" are assumed to be package names, except for username and password
+    def __init__(self, value):
+        if value == self.DEFAULT:
+            value = getpass.getpass("UniVie u:space password: ")
+        self.value = value
 
+    def __repr__(self):
+        return f"{self.__class__.__name__}(*****)"
 
-Publish development versions
-----------------------------
-To compile and upload a development version, use the cdev tag
+    def __str__(self):
+        return self.value
 
-$ python irdb/publish.py -cdev -u <PKG_NAME> ... <PKG_NAME_N> -l <USERNAME> -p <PASSWORD>
-
-"""
-
-
-with open(PATH_PACKAGES_YAML, "r",
-          encoding="utf8") as f_pkgs:
-    PKGS = yaml.full_load(f_pkgs)
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            raise TypeError()
+        return str(self) == str(other)
```

Comment:
To make `__eq__()` work with my suggestion to remove `__str__()`
```suggestion
        return self.value == other.value
```